test/AWSCredentials.jl

```
@@ -0,0 +1,549 @@
+macro test_ecode(error_codes, expr)
+    quote
+        try
+            $expr
+            @test false
+        catch e
+            if e isa AWSException
+                ecode(e) in [$error_codes;]
+            end
```

Should we rethrow the error if it wasn't an AWSException?